syntax = "proto3";

package database;

message Empty {
}

message CallStatus {
  bool success = 1;
}

message DatabasePath {
  string path = 1;
}

service Database {
  rpc CheckDatabaseExistence(Empty) returns (CallStatus);
  rpc GetDatabasePath(Empty) returns (DatabasePath);
  rpc SetDatabasePath(DatabasePath) returns (CallStatus);
  rpc CreateIfNotExists(Empty) returns (CallStatus);
  rpc DeleteDatabase(Empty) returns (CallStatus);
}
